src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1203,11 +1203,13 @@ public function getMethod(): string
 
         $method = strtoupper($method);
 
-        if (\in_array($method, ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'PATCH', 'PURGE', 'TRACE', 'QUERY'], true)) {
+        if (\in_array($method, ['GET', 'HEAD', 'CONNECT', 'TRACE'], true)) {
+            trigger_deprecation('symfony/http-foundation', '7.4', 'HTTP method override is deprecated for methods GET, HEAD, CONNECT and TRACE; it will be ignored in Symfony 8.0.', $method);
```

Comment:
Without a setting to disable it for projects that don't intent to use this feature (maybe even a setting to disable support for `X-HTTP-METHOD-OVERRIDE` btw), this deprecation is not actionable for projects that are exposed to uncontrolled user agents (i.e. 99.9999% of Symfony projects based on a non-scientific guess) as any incoming request could add such header (even when the project did not intend to rely on it).

Reply:
I'd prefer not adding a setting for this. HTTP method overriding is not really for public API needs anyway.

Reply:
That's actually one more reason to allow projects to disable support for this header entirely if they don't rely on it internally.
With the header being always supported, it actually exposes it in the public API of all Symfony-based projects.

Note that even though Rails has the feature enabled by default (and without our setting to support only the header and not the parameter), its API allowing to reconfigure the Rack middlewares that are registered by default means that projects can remove the `Rack::MethodOverride` middleware if they don't want to support the feature in their project.

Reply:
I'm looking for less things to configure. I think there's no value adding a config option for this, even to adopt the 8.0-behavior ahead of time. Yes, this means the outside can trigger deprecations at will. That's also quite often possible in many other ways so I wouldn't anticipate an issue for this one. KISS.

Adding an option would mean documenting it, some ppl adopting it, then deprecating it because we won't want to keep yet another option for something that does not harm if unconfigurable (after hardening it of course, as prepared here), then ppl removing it, and us removing it again in 9.0. This feels too costly to me.

Reply:
I figured out a way, see #61979